src/hivpy/hiv_status.py

```diff
@@ -24,17 +39,54 @@ def initial_HIV_status(self, population: pd.DataFrame):
         hiv_status = self._prob_HIV_initial(population[col.AGE]) > rng.random(len(population))
         return hiv_status
 
-    def update_HIV_status(self, population: pd.DataFrame):
+    def update_partner_risk_vectors(self, population):
+        """calculate the risk factor associated with each sex and age group"""
+        # Should we be using for loops here or can we do better? 
+        for sex in SexType:
+            for age_group in range(5):   # FIXME need to get number of age groups from somewhere
+                sub_pop_indices = population.data.index[(population.data[col.SEX]==sex) & (population.data[col.SEX_MIX_AGE_GROUP] == age_group)]
+                sub_pop = population.data.loc[sub_pop_indices]
```

Comment:
I think we can get this in one step?
```suggestion
                sub_pop = population.data.loc[(population.data[col.SEX]==sex) & (population.data[col.SEX_MIX_AGE_GROUP] == age_group)]
```